src/measure_theory/vector_measure.lean

```diff
@@ -0,0 +1,382 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.integration
+
+/-!
+
+# Vector valued measures
+
+This file defines vector valued measures, which are σ-additive functions from a set to a add monoid
+`M` such that it maps the empty set and non-measurable sets to zero. In the case
+that `M = ℝ`, we called the vector measure a signed measure and write `signed_measure α`.
+Similarly, when `M = ℂ`, we call the measure a complex measure and write `complex_measure α`.
+
+## Main definitions
+
+* `vector_measure` is a vector valued, σ-additive function that maps the empty
+  and non-measurable set to zero.
+
+## Implementation notes
+
+We require all non-measurable sets to be mapped to zero in order for the extensionality lemma
+to only compare the underlying functions for measurable sets.
+
+We use `has_sum` instead of `tsum` in the definition of vector measures in comparison to `measure`
+since this provides summablity.
+
+## Tags
+
+vector measure, signed measure, complex measure
+-/
+
+noncomputable theory
+
+open_locale classical big_operators nnreal ennreal
+
+namespace measure_theory
+
+variables {α β : Type*} [measurable_space α]
+
+/-- A vector measure on a measurable space `α` is a σ-additive `M`-valued function (for some `M`
+an add monoid) such that the empty set and non-measurable sets are mapped to zero. -/
+structure vector_measure (α : Type*) [measurable_space α]
+  (M : Type*) [add_comm_monoid M] [topological_space M] :=
+(measure_of : set α → M)
+(empty : measure_of ∅ = 0)
+(not_measurable ⦃i : set α⦄ : ¬ measurable_set i → measure_of i = 0)
+(m_Union ⦃f : ℕ → set α⦄ :
+  (∀ i, measurable_set (f i)) → pairwise (disjoint on f) →
+  has_sum (λ i, measure_of (f i)) (measure_of (⋃ i, f i)))
+
+-- A `signed_measure` is a `ℝ`-vector measure.
+notation `signed_measure ` α := vector_measure α ℝ
+-- A `complex_measure` is a `ℂ`-vector_measure.
+notation `complex_measure ` α := vector_measure α ℂ
```

Comment:
Possibly better as
```suggestion
/-- A `signed_measure` is a `ℝ`-vector measure. -/
abbreviation signed_measure (α : Type*) [measurable_space α] := vector_measure α ℝ
/-- A `complex_measure` is a `ℂ`-vector_measure. -/
abbreviation complex_measure (α : Type*) [measurable_space α] := vector_measure α ℂ
```

The main advantages of this is that the docstring will show up in doc-gen, and you can define theorems like `signed_measure.foo` which will work with dot notation. The disadvantage is that `vector_measure α ℝ` will not automatically display in a goal view as `signed_measure α`.

I'll let @sgouezel decide whether this is a good idea.

Reply:
The dot notation argument sounds a good one to me.